spec/unit/face/config_spec.rb

```diff
@@ -44,6 +44,53 @@ module PuppetFaceSpecs
     subject.print('all')
   end
 
+  context "when setting config values" do
+    let(:config_file) { '/foo/puppet.conf' }
+    let(:path) { Pathname.new(config_file) }
+    before(:each) do
+      Puppet[:config] = '/foo/puppet.conf'
+      Puppet::FileSystem.stubs(:pathname).with(config_file).returns(path)
+      Puppet::FileSystem.stubs(:touch)
+    end
+
+    it "writes to the correct puppet config file" do
+      Puppet::FileSystem.expects(:open).with(path, anything, anything)
+      subject.set('foo', 'bar')
+    end
+
+    it "creates a config file if one does not exist" do
```

Comment:
This test is a little misleading (especially with the expectations written in this order).

That a file gets created is a side effect of the `touch` call - but the problem here with the test is that there's no assertion that the file didn't exist prior to calling `subject.set`